torch/_inductor/config.py

```diff
@@ -208,150 +219,153 @@ class cpp:
-    persistent_reductions = (
+    persistent_reductions: bool = (
         os.environ.get("TORCHINDUCTOR_PERSISTENT_REDUCTIONS", "1") == "1"
     )
 
     # hint to Triton when arguments are divisible by 16
-    divisible_by_16 = True
+    divisible_by_16: bool = True
 
     # theses are not enforced, but they are used by asserts in triton_heuristics.py
     # NOTE: mobilevit_s in timm_models required X to be set to the higher value 2048
-    max_block = {"X": 2048, "Y": 1024, "Z": 1024}
+    max_block: Dict[str, int] = dataclasses.field(
+        default_factory=lambda: {"X": 2048, "Y": 1024, "Z": 1024}
+    )
 
     # Store the generated cubin files for cpp wrapper code to load
-    store_cubin = False
+    store_cubin: bool = False
 
     # the max number of spills we allow for the configs we benchmark.
     # Setting this to 0 means we skip a config if it spills even a single
     # register.
     # Settting it to a larger value allows a config spilling a small amount
     # of registers being benchmarked.
-    spill_threshold = 0
+    spill_threshold: int = 0
 
     # Inject a bug into our relu implementation; useful for testing our repro
     # extraction and minification functionality.
     # Valid values: "compile_error", "runtime_error", "accuracy"
-    inject_relu_bug_TESTING_ONLY = None
+    inject_relu_bug_TESTING_ONLY: Optional[str] = None
 
 
 # create a directory containing lots of debug information
-class trace:
+@dataclasses.dataclass
+class TraceConfig(ConfigMixin):
     # master switch for all debugging flags below
-    enabled = os.environ.get("TORCH_COMPILE_DEBUG", "0") == "1"
+    enabled: bool = os.environ.get("TORCH_COMPILE_DEBUG", "0") == "1"
 
     # Save python logger call >=logging.DEBUG
-    debug_log = False
+    debug_log: bool = False
 
     # Save python logger call >=logging.INFO
-    info_log = False
+    info_log: bool = False
 
     # Save input FX graph (post decomps, pre optimization)
-    fx_graph = True
+    fx_graph: bool = True
 
     # Save FX graph after transformations
-    fx_graph_transformed = True
+    fx_graph_transformed: bool = True
 
     # Save TorchInductor IR before fusion pass
-    ir_pre_fusion = True
+    ir_pre_fusion: bool = True
 
     # Save TorchInductor IR after fusion pass
-    ir_post_fusion = True
+    ir_post_fusion: bool = True
 
     # Copy generated code to trace dir
-    output_code = True
+    output_code: bool = True
 
     # SVG figure showing post-fusion graph
-    graph_diagram = False
+    graph_diagram: bool = False
 
     # Store cProfile (see snakeviz to view)
-    compile_profile = False
+    compile_profile: bool = False
 
     # Upload the .tar.gz file
     # Needs to be overriden based on specific environment needs
-    upload_tar = None
-
+    upload_tar: Any = dataclasses.field(default=None, metadata={"skip_pickle": True})
 
-_save_config_ignore = {
-    # workaround: "Can't pickle <function ...>"
-    "trace.upload_tar",
-}
```

Comment:
Won't we still need this workaround?  I believe this was an fbcode issue.

Reply:
pickle will try to pickle the `trace` object, which will then call `__getstate__` on it, `__getstate__` defined by ConfigMixin will ignore fields marked with skip_pickle.

I added a test to make sure.